File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.Arrays;

class ResistorColorDuo {
    int value(String[] colors) {
        String[] color = new String[] {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
        String Color1 = colors[0];
        String Color2 = colors[1];
        return Arrays.asList(color).indexOf(Color1) * 10 + Arrays.asList(color).indexOf(Color2);
        
    }
}

File: solutions/c/resistor-color-duo/1/resistor_color_duo.c
#include "resistor_color_duo.h"

int one_color_code(resistor_band_t color) {
    if (color == BLACK)
        return 0;
    else if (color == BROWN)
        return 1;
    else if (color == RED)
        return 2;
    else if (color == ORANGE)
        return 3;
    else if (color == YELLOW)
        return 4;
    else if (color == GREEN)
        return 5;
    else if (color == BLUE)
        return 6;
    else if (color == VIOLET)
        return 7;
    else if (color == GREY)
        return 8;
    else 
        return 9;   
}

int color_code(const resistor_band_t array[] ) {
    return 10 * one_color_code(array[0]) +  one_color_code(array[1]);
}

